### DISCUSSION TITLE
Installation & configuration

### DISCUSSION THREAD

**Discussion by Pavan0520** [sipcapture/homer#650] [Q&A] [answered]
Hi all. I installed homer10 with docker. I am new to Homer10. How can I confirm my server to Homer? Can somebody assist me with correctly installing and configuring my server?

**Comment by adubovikov** [comment, marked as answer]
Hi [@Pavan0520](https://github.com/Pavan0520),

Welcome. A few clarifications first, then a concrete checklist.

### Homer 10 vs Homer 11

If you installed **Homer 10 preview docker**, that stack was experimental (Grafana/qryn oriented).

For a **new install today**, use **Homer 11**:

- Releases: https://github.com/sipcapture/homer/releases
- Docker: `examples/docker` in this repo
- Image: `ghcr.io/sipcapture/homer:latest`

Homer 11 is the supported all-in-one stack with the bundled web UI for SIP search and call flow.

### What "confirm my server to Homer" means

Your PBX/SBC/agent does not register with Homer like a SIP endpoint. You **send HEPv3** to Homer ingest:

| Direction | What to configure |
|-----------|-------------------|
| Asterisk | `hep.conf` + `res_hep` / `res_hep_pjsip` → Homer IP:9060 |
| Kamailio/OpenSIPS | HEP module / siptrace → Homer IP:9060 |
| heplify / hepagent | `-hs HomerIP:9060` or transport config |
| Oracle SBC / other | HEP export to UDP/TCP **9060** on Homer |

Default ingest ports in Homer 11 docker:

- **UDP/TCP 9060** — HEPv3
- **HTTP 9080** — optional HTTP ingest
- **8080** — web UI + API

### Quick install (Homer 11 docker)

```bash
git clone https://github.com/sipcapture/homer.git
cd homer/examples/docker
docker compose -f docker-compose.yaml up -d
```

Open: http://localhost:8080

Default login (change after first login): **`admin` / `sipcapture`**

See: `examples/docker/README.md`

### Verify Homer is receiving data

**1. Container healthy**

```bash
docker compose ps
docker logs homer-docker-homer-1 2>&1 | tail -30
```

**2. Ingest port listening**

```bash
ss -ulnp | grep 9060
ss -tlnp | grep 9060
```

**3. Packets arriving (on Homer host)**

```bash
tcpdump -ni any udp port 9060 -c 10
```

If this shows nothing, the sender is not reaching Homer (firewall, wrong IP/port, NAT).

**4. Data visible in UI**

- Log in at `:8080`
- Open **Search**
- Time range: last 15 minutes
- Search with empty/minimal filters or filter by your source IP

**5. CLI check (optional)**

```bash
homer search --host localhost:8080 --user admin --pass sipcapture --from 15m --method INVITE
```

**6. Metrics (optional)**

```bash
curl -s localhost:9090/metrics | grep -i hep
```

### Common misconfigurations

| Symptom | Likely cause |
|---------|----------------|
| UI works, no calls | Agent points to wrong IP/port; UDP 9060 blocked |
| tcpdump sees packets, UI empty | Wrong capture ID / timezone / search window too narrow |
| Worked once, stopped after reboot | systemd ordering or firewall (see also discussion #629) |
| H10 docker, confused UI | H10 preview is not H11; switch to `examples/docker` |

### Agent-side example (heplify)

```bash
heplify -hs HOMER_IP:9060 -dim SIP -nt UDP
```

Mirror SIP from a span port or run on the SBC if it supports HEP export.

### Next step if still stuck

Reply with:

1. Homer version (H10 preview vs H11 release tag)
2. What sends HEP (Asterisk, SBC, heplify, etc.)
3. Output of `tcpdump -ni any udp port 9060 -c 5`
4. Whether http://localhost:8080 login works

We can narrow it down from there.
